### PR TITLE
ISPN-13072 Treat all HTTP authentication failures as such

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-13072

### CODE REVIEW

**Comment by pruivo** [comment]
@tristantarrant checkstyle didn't like it

File: server/tests/src/test/java/org/infinispan/server/security/authentication/RestAuthentication.java [hunk 1]
```diff
@@ -1,13 +1,22 @@
 package org.infinispan.server.security.authentication;
 
+import static com.mysql.cj.util.StringUtils.getBytes;
```
Comment (pruivo) [inline server/tests/src/test/java/org/infinispan/server/security/authentication/RestAuthentication.java:3]: is this intentional? using `com.mysql.*` ?

**Comment by tristantarrant** [comment]
I cherry-picked from the wrong branch

**Comment by tristantarrant** [comment]
@pruivo repushed

**Comment by pruivo** [comment]
checkstyle still failing :)

**Comment by pruivo** [comment]
integrated! thanks @tristantarrant !

**Pull request merged**